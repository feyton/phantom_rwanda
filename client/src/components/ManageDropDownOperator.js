import { axiosBase as axios } from '@utils/Api.js';
import React, { useRef, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import Swal from 'sweetalert2';
import CheckRole from '../utils/CheckRoles.js';
import { ButtonLoading } from './Button.js';

function ManageDropDownOperator({ row }) {
  const { user, mobileNumber, nationalID, company, id } = row.original;
  const navigate = useNavigate();
  const selectRef = useRef();
  const [loading, setloading] = useState(false);
  const handleEdit = () => {
    toast('Clicked');
  };

  const data = {
    name: user.firstName,
    email: user.email,
    mobileNumber,
    nationalID,
    company
  };
  const handlePermissions = () => {
    selectRef.current.value = 'manage';
    navigate(`/dashboard/modal/permission/change/${nationalID}`, {
      state: data
    });
  };

  const handleDelete = async () => {
    selectRef.current.value = 'manage';
    Swal.fire({
      title: 'Are you sure?',
      html: `Operator <b>${data.name}</b> will be deleted.`,
      text: 'This operator will be deleted!',
      icon: 'warning',
      showCancelButton: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
      confirmButtonText: 'Yes, delete!'
    }).then(async (result) => {
      if (result.isConfirmed) {
        setloading(true);
        /* istanbul ignore next */
        try {
          await axios.delete(`/operators/${id}`);
          setloading(false);
          selectRef.current.innerHTML = `<b className="deleted mx-auto font-bold">Deleted</>`;
        } catch (error) {
          toast('Something went wrong');
          setloading(false);
        }
      }
    });
  };
  const handleChange = (e) => {
    switch (e.target.value) {
      case 'perm':
        return handlePermissions();
      case 'edit':
        return handleEdit();
      case 'delete':
        return handleDelete();
      default:
        break;
    }
  };

  return (
    <div className="z-10 font-sm" ref={selectRef}>
      {loading ? (
        <ButtonLoading name="Sending..." />
      ) : (
        <select
          onChange={handleChange}
          name=""
          id="select"
          role="select"
          className="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        >
          <option hidden value="manage">
            Manage
          </option>
          <option value="view">View</option>
          <CheckRole
            children={
              <option
                className="cursor-pointer font-raleway font-bold"
                value="edit"
              >
                Edit
              </option>
            }
            role={['admin']}
          />
          <CheckRole
            children={
              <option className="cursor-pointer" value="delete">
                Delete
              </option>
            }
            role={['admin']}
          />

          <CheckRole
            children={
              <option className="cursor-pointer" value="perm">
                Permissions
              </option>
            }
            role={['admin']}
          />
        </select>
      )}
    </div>
  );
}

export default ManageDropDownOperator;
