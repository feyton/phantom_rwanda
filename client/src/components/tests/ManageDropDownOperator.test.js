import { fireEvent, render, screen } from '@testing-library/react';
import ProviderWrapper from '@utils/TestUtil.js';
import React from 'react';
import * as redux from 'react-redux';
import Swal from 'sweetalert2';
import ManageDropDownOperator from '../ManageDropDownOperator.js';

let spyOnUseSelector;

const mockUseLocationValue = {
  pathname: '/testroute',
  search: '',
  hash: '',
  state: null
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockImplementation(() => {
    return mockUseLocationValue;
  })
}));

const rowData = {
  original: {
    user: {
      firstName: 'Fabrice',
      email: ' test@me.com'
    },
    license: 199700,
    assigned_bus: null,
    id: 1
  }
};

const swalSpy = jest.spyOn(Swal, 'fire');

describe('Manage Dropdown', () => {
  spyOnUseSelector = jest.spyOn(redux, 'useSelector');
  spyOnUseSelector.mockReturnValue({
    authenticated: true,
    user: {
      firstName: 'Fabrice',
      role: 'admin'
    }
  });

  it('Should render the dropdown', async () => {
    const elem = render(
      <ProviderWrapper>
        <ManageDropDownOperator row={rowData} />
      </ProviderWrapper>
    );
    expect(elem).toMatchSnapshot();
  });
  it('Should call the edit', async () => {
    const elem = render(
      <ProviderWrapper>
        <ManageDropDownOperator row={rowData} />
      </ProviderWrapper>
    );
    fireEvent.change(screen.getByTestId('select'), {
      target: {
        value: 'edit'
      }
    });
    expect(elem).toMatchSnapshot();
  });
  it('Should call the assign', async () => {
    const elem = render(
      <ProviderWrapper>
        <ManageDropDownOperator row={rowData} />
      </ProviderWrapper>
    );
    fireEvent.change(screen.getByTestId('select'), {
      target: {
        value: 'assign'
      }
    });
    expect(elem).toMatchSnapshot();
  });
  it('Should call the permissions', async () => {
    const elem = render(
      <ProviderWrapper>
        <ManageDropDownOperator row={rowData} />
      </ProviderWrapper>
    );
    fireEvent.change(screen.getByTestId('select'), {
      target: {
        value: 'perm'
      }
    });
    expect(elem).toMatchSnapshot();
  });
  it('Should call the delete option', async () => {
    const elem = render(
      <ProviderWrapper>
        <ManageDropDownOperator row={rowData} />
      </ProviderWrapper>
    );
    fireEvent.change(screen.getByTestId('select'), {
      target: {
        value: 'delete'
      }
    });
    expect(elem).toMatchSnapshot();
    fireEvent.click(screen.getByText(/Yes, delete!/));
    expect(elem).toMatchSnapshot();
  });
  it('Should render the dropdown with change bus', async () => {
    rowData.original.assigned_bus = 'fabrice';
    const elem = render(
      <ProviderWrapper>
        <ManageDropDownOperator row={rowData} />
      </ProviderWrapper>
    );
    fireEvent.change(screen.getByTestId('select'), {
      target: {
        value: 'nothing'
      }
    });
    expect(elem).toMatchSnapshot();
  });
});
