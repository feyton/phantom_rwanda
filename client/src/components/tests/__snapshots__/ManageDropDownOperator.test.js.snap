// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage Dropdown Should call the assign 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10 font-sm"
      >
        <select
          class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
          id="select"
          name=""
          role="select"
        >
          <option
            hidden=""
            value="manage"
          >
            Manage
          </option>
          <option
            value="view"
          >
            View
          </option>
          <option
            class="cursor-pointer font-raleway font-bold"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="perm"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10 font-sm"
    >
      <select
        class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        id="select"
        name=""
        role="select"
      >
        <option
          hidden=""
          value="manage"
        >
          Manage
        </option>
        <option
          value="view"
        >
          View
        </option>
        <option
          class="cursor-pointer font-raleway font-bold"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="perm"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Manage Dropdown Should call the delete option 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="swal2-shown swal2-height-auto"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="z-10 font-sm"
      >
        <select
          class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
          id="select"
          name=""
          role="select"
        >
          <option
            hidden=""
            value="manage"
          >
            Manage
          </option>
          <option
            value="view"
          >
            View
          </option>
          <option
            class="cursor-pointer font-raleway font-bold"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="perm"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
    <div
      class="swal2-container swal2-center swal2-backdrop-show"
    >
      <div
        aria-describedby="swal2-html-container"
        aria-labelledby="swal2-title"
        aria-live="assertive"
        aria-modal="true"
        class="swal2-popup swal2-modal swal2-icon-warning"
        role="dialog"
        style="opacity: 0 !important; display: grid;"
        tabindex="-1"
      >
        <button
          aria-label="Close this dialog"
          class="swal2-close"
          style="display: none;"
          type="button"
        >
          Ã—
        </button>
        <ul
          class="swal2-progress-steps"
          style="display: none;"
        />
        <div
          class="swal2-icon swal2-warning swal2-icon-show"
          style="display: flex;"
        >
          <div
            class="swal2-icon-content"
          >
            !
          </div>
        </div>
        <img
          class="swal2-image"
          style="display: none;"
        />
        <h2
          class="swal2-title"
          id="swal2-title"
          style="display: block;"
        >
          Are you sure?
        </h2>
        <div
          class="swal2-html-container"
          id="swal2-html-container"
          style="display: block;"
        >
          Operator 
          <b>
            Fabrice
          </b>
           will be deleted.
        </div>
        <input
          class="swal2-input"
          style="display: none;"
        />
        <input
          class="swal2-file"
          style="display: none;"
          type="file"
        />
        <div
          class="swal2-range"
          style="display: none;"
        >
          <input
            type="range"
          />
          <output />
        </div>
        <select
          class="swal2-select"
          style="display: none;"
        />
        <div
          class="swal2-radio"
          style="display: none;"
        />
        <label
          class="swal2-checkbox"
          for="swal2-checkbox"
          style="display: none;"
        >
          <input
            type="checkbox"
          />
          <span
            class="swal2-label"
          />
        </label>
        <textarea
          class="swal2-textarea"
          style="display: none;"
        />
        <div
          class="swal2-validation-message"
          id="swal2-validation-message"
          style="display: none;"
        />
        <div
          class="swal2-actions"
          style="display: flex;"
        >
          <div
            class="swal2-loader"
          />
          <button
            aria-label=""
            class="swal2-confirm swal2-styled swal2-default-outline"
            style="display: inline-block; background-color: rgb(48, 133, 214);"
            type="button"
          >
            Yes, delete!
          </button>
          <button
            aria-label=""
            class="swal2-deny swal2-styled"
            style="display: none;"
            type="button"
          >
            No
          </button>
          <button
            aria-label=""
            class="swal2-cancel swal2-styled swal2-default-outline"
            style="display: inline-block; background-color: rgb(221, 51, 51);"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div
          class="swal2-footer"
          style="display: none;"
        />
        <div
          class="swal2-timer-progress-bar-container"
        >
          <div
            class="swal2-timer-progress-bar"
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="z-10 font-sm"
    >
      <select
        class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        id="select"
        name=""
        role="select"
      >
        <option
          hidden=""
          value="manage"
        >
          Manage
        </option>
        <option
          value="view"
        >
          View
        </option>
        <option
          class="cursor-pointer font-raleway font-bold"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="perm"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Manage Dropdown Should call the delete option 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class=""
  >
    <div>
      <div
        class="z-10 font-sm"
      >
        <select
          class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
          id="select"
          name=""
          role="select"
        >
          <option
            hidden=""
            value="manage"
          >
            Manage
          </option>
          <option
            value="view"
          >
            View
          </option>
          <option
            class="cursor-pointer font-raleway font-bold"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="perm"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10 font-sm"
    >
      <select
        class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        id="select"
        name=""
        role="select"
      >
        <option
          hidden=""
          value="manage"
        >
          Manage
        </option>
        <option
          value="view"
        >
          View
        </option>
        <option
          class="cursor-pointer font-raleway font-bold"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="perm"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Manage Dropdown Should call the edit 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10 font-sm"
      >
        <select
          class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
          id="select"
          name=""
          role="select"
        >
          <option
            hidden=""
            value="manage"
          >
            Manage
          </option>
          <option
            value="view"
          >
            View
          </option>
          <option
            class="cursor-pointer font-raleway font-bold"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="perm"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10 font-sm"
    >
      <select
        class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        id="select"
        name=""
        role="select"
      >
        <option
          hidden=""
          value="manage"
        >
          Manage
        </option>
        <option
          value="view"
        >
          View
        </option>
        <option
          class="cursor-pointer font-raleway font-bold"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="perm"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Manage Dropdown Should call the permissions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10 font-sm"
      >
        <select
          class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
          id="select"
          name=""
          role="select"
        >
          <option
            hidden=""
            value="manage"
          >
            Manage
          </option>
          <option
            value="view"
          >
            View
          </option>
          <option
            class="cursor-pointer font-raleway font-bold"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="perm"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10 font-sm"
    >
      <select
        class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        id="select"
        name=""
        role="select"
      >
        <option
          hidden=""
          value="manage"
        >
          Manage
        </option>
        <option
          value="view"
        >
          View
        </option>
        <option
          class="cursor-pointer font-raleway font-bold"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="perm"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Manage Dropdown Should render the dropdown 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="z-10 font-sm"
      >
        <select
          class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
          id="select"
          name=""
          role="select"
        >
          <option
            hidden=""
            value="manage"
          >
            Manage
          </option>
          <option
            value="view"
          >
            View
          </option>
          <option
            class="cursor-pointer font-raleway font-bold"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="perm"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10 font-sm"
    >
      <select
        class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        id="select"
        name=""
        role="select"
      >
        <option
          hidden=""
          value="manage"
        >
          Manage
        </option>
        <option
          value="view"
        >
          View
        </option>
        <option
          class="cursor-pointer font-raleway font-bold"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="perm"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Manage Dropdown Should render the dropdown with change bus 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class=""
  >
    <div>
      <div
        class="z-10 font-sm"
      >
        <select
          class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
          id="select"
          name=""
          role="select"
        >
          <option
            hidden=""
            value="manage"
          >
            Manage
          </option>
          <option
            value="view"
          >
            View
          </option>
          <option
            class="cursor-pointer font-raleway font-bold"
            value="edit"
          >
            Edit
          </option>
          <option
            class="cursor-pointer"
            value="delete"
          >
            Delete
          </option>
          <option
            class="cursor-pointer"
            value="perm"
          >
            Permissions
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="z-10 font-sm"
    >
      <select
        class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
        id="select"
        name=""
        role="select"
      >
        <option
          hidden=""
          value="manage"
        >
          Manage
        </option>
        <option
          value="view"
        >
          View
        </option>
        <option
          class="cursor-pointer font-raleway font-bold"
          value="edit"
        >
          Edit
        </option>
        <option
          class="cursor-pointer"
          value="delete"
        >
          Delete
        </option>
        <option
          class="cursor-pointer"
          value="perm"
        >
          Permissions
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
